core/src/main/java/org/wildfly/channel/spi/MavenVersionsResolver.java

```diff
@@ -70,6 +70,16 @@ public interface MavenVersionsResolver extends Closeable {
     */
    List<File> resolveArtifacts(List<ArtifactCoordinate> coordinates) throws UnresolvedMavenArtifactException;
 
+   /**
+    * Returns the latest {@code release} version according to the repositorie's Maven metadata.
+    *
+    * @param groupId Maven GroupId - required
+    * @param artifactId Maven ArtifactId - required
+    *
+    * @return the set of versions.
+    */
+   public String getReleaseVersion(String groupId, String artifactId);
```

Comment:
for the latest version, we should also use the `<latest>` field from the `maven-metadata.xml`. This would be more consistent and make it clear that we delegate to Maven repository to figure out which is the latest version.